test/matchers/element/toHaveText.test.ts

```diff
@@ -104,6 +104,66 @@ describe('toHaveText', () => {
         expect(result.pass).toBe(true)
     })
 
+    test('should return true if actual text + single replacer matches the expected text', async () => {
+        const el: any = await $('sel')
+        el._text = function (): string {
+            return 'WebdriverIO'
+        }
+
+        const result = await toHaveText.bind({})(el, 'BrowserdriverIO', { replace: ['Web', 'Browser'] })
+        expect(result.pass).toBe(true)
+    })
+
+    test('should return true if actual text + replace (string) matches the expected text', async () => {
+        const el: any = await $('sel')
+        el._text = function (): string {
+            return 'WebdriverIO'
+        }
+
+        const result = await toHaveText.bind({})(el, 'BrowserdriverIO', { replace: { replaceWebWithBrowser: ['Web', 'Browser'] } })
```

Comment:
I am not sure if I understand this correctly, where does `replaceWebWithBrowser` come from and what does it mean?

Reply:
Since you mentioned you wanted an object as replacer this is how I implemented it.

You can name your individual replacers but also support a single replacer.

This looks like so:
```js
toHaveText({ replace: {
  myCustomReplacer: ['foo', 'bar']
  // can add more replacers
} })
```
and
```js
toHaveText({ replace: ['foo', 'bar'] })
```

Both work but I can also change the object to be an array in which case it would be:
```js
toHaveText({ replace: [
  ['foo', 'bar']
  // can add more replacers
]})
```
and
```js
toHaveText({ replace: ['foo', 'bar'] })
```